src/NuGet.Clients/NuGet.PackageManagement.UI/ViewModels/ReadMePreviewViewModel.cs

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Threading.Tasks;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+using NuGet.VisualStudio.Telemetry;
+
+namespace NuGet.PackageManagement.UI.ViewModels
+{
+    public sealed class ReadMePreviewViewModel : ViewModelBase
+    {
+        public ReadMePreviewViewModel()
+        {
+            _isReadMeAvailable = false;
+            _isErrorWithReadMe = false;
+            _rawReadMe = string.Empty;
+        }
+
+        private bool _isErrorWithReadMe;
+
+        public bool IsErrorWithReadMe
+        {
+            get => _isErrorWithReadMe;
+            set
+            {
+                _isErrorWithReadMe = value;
+                RaisePropertyChanged(nameof(IsErrorWithReadMe));
+            }
+        }
+
+        private bool _isReadMeAvailable;
+
+        public bool IsReadMeAvailable
+        {
+            get => _isReadMeAvailable;
+            set
+            {
+                _isReadMeAvailable = value;
+                RaisePropertyChanged(nameof(IsReadMeAvailable));
+            }
+        }
+
+        private string _rawReadMe;
+
+        public string ReadMeMarkdown
+        {
+            get => _rawReadMe;
+            set
+            {
+                _rawReadMe = value;
+                RaisePropertyChanged(nameof(ReadMeMarkdown));
+            }
+        }
+
+        public async Task UpdateReadMe(string packagePath, string packageId)
```

Comment:
nit: the term update makes me think of writing content to disk.
Reload or Load, maybe?

```suggestion
        public async Task LoadReadMe(string packagePath, string packageId)
```